src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultOutputPaths.targets

```diff
@@ -11,29 +11,109 @@ Copyright (c) .NET Foundation. All rights reserved.
 -->
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-   <!--
-    Apply the same default output paths as Microsoft.Common.targets now since we're running before them,
-    but need to adjust them and/or make decisions in terms of them.
-
-    Also note that common targets only set a default OutputPath if neither configuration nor
+  <!--
+    Note that common targets only set a default OutputPath if neither configuration nor
     platform were set by the user. This was used to validate that a valid configuration is passed,
     assuming the convention maintained by VS that every Configuration|Platform combination had
     an explicit OutputPath. Since we now want to support leaner project files with less
-    duplication and more automatic defaults, we always set a default OutputPath and can no
-    longer depend on that convention for validation. Getting validation re-enabled with a
-    different mechanism is tracked by https://github.com/dotnet/sdk/issues/350
+    duplication and more automatic defaults, we always set a default OutputPath.
    -->
-  <PropertyGroup>
-    <Configuration Condition="'$(Configuration)'==''">Debug</Configuration>
-    <Platform Condition="'$(Platform)'==''">AnyCPU</Platform>
-    <PlatformName Condition="'$(PlatformName)' == ''">$(Platform)</PlatformName>
 
+  <!-- Projects which don't use Microsoft.NET.Sdk will typically define the OutputPath directly (usually in a
+       Configuration-specific PropertyGroup), so in that case we won't append to it by default. -->
+  <PropertyGroup Condition="'$(UsingNETSdkDefaults)' == 'true'">
+    <AppendTargetFrameworkToOutputPath Condition="'$(AppendTargetFrameworkToOutputPath)' == ''">true</AppendTargetFrameworkToOutputPath>
+  </PropertyGroup>
+
+  <!-- NOTE: If we want to default UseArtifactsOutput to true when targeting a given version of .NET or higher, this is where we would do it.
+
+    It would look something like this:
+
+  <PropertyGroup Condition="'$(UseArtifactsOutput)' == '' and
+                            '$(TargetFrameworks)' == '' and
+                            '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                            $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 8.0))">
+    <UseArtifactsOutput>true</UseArtifactsOutput>
+  </PropertyGroup>
+  -->
+
+  <!-- Handle ArtifactsPath if it wasn't set in Sdk.props -->
+  <PropertyGroup Condition="'$(ArtifactsPath)' != ''">
+    <UseArtifactsOutput Condition="'$(UseArtifactsOutput)' == ''">true</UseArtifactsOutput>
+  </PropertyGroup>
+
+  <!-- Repeat ArtifactsPath logic from Sdk.props here, in case UseArtifactsOutput was set in the project file -->
```

Comment:
Could we error in that case instead? I don't like the idea of accidentally diverging the implementation over time.

Reply:
So `ArtifactsPath` just wouldn't be allowed in the project file at all?

We go through some hoops to make `BaseIntermediateOutputPath` work whether it's set in the project file or beforehand.  If it's set in the project file then it doesn't affect `MSBuildProjectExtensionsPath`.  This seemed analogous.

I just had a crazy idea: Maybe we could de-duplicate some of the logic by having a .props file that's conditionally imported in two different places depending on where `ArtifactsPath` is defined?